game/scrabble.js

```diff
@@ -1,103 +1,86 @@
-// var boardMap = [[32, 1, 1, 21, 1, 1, 1, 32, 1, 1, 1, 21, 1, 1, 32],
-//                 [1, 22, 1, 1, 1, 31, 1, 1, 1, 31, 1, 1, 1, 22, 1],
-//                 [1, 1, 22, 1, 1, 1, 21, 1, 21, 1, 1, 1, 22, 1, 1],
-//                 [21, 1, 1, 22, 1, 1, 1, 21, 1, 1, 1, 22, 1, 1, 21],
-//                 [1, 1, 1, 1, 22, 1, 1, 1, 1, 1, 22, 1, 1, 1, 1],
-//                 [1, 31, 1, 1, 1, 31, 1, 1, 1, 31, 1, 1, 1, 31, 1],
-//                 [1, 1, 21, 1, 1, 1, 21, 1, 21, 1, 1, 1, 21, 1, 1],
-//                 [32, 1, 1, 21, 1, 1, 1, 99, 1, 1, 1, 21, 1, 1, 32],
-//                 [1, 1, 21, 1, 1, 1, 21, 1, 21, 1, 1, 1, 21, 1, 1],
-//                 [1, 31, 1, 1, 1, 31, 1, 1, 1, 31, 1, 1, 1, 31, 1],
-//                 [1, 1, 1, 1, 22, 1, 1, 1, 1, 1, 22, 1, 1, 1, 1],
-//                 [21, 1, 1, 22, 1, 1, 1, 21, 1, 1, 1, 22, 1, 1, 21],
-//                 [1, 1, 22, 1, 1, 1, 21, 1, 21, 1, 1, 1, 22, 1, 1],
-//                 [1, 22, 1, 1, 1, 31, 1, 1, 1, 31, 1, 1, 1, 22, 1],
-//                 [32, 1, 1, 21, 1, 1, 1, 32, 1, 1, 1, 21, 1, 1, 32]]
-// boardMap.forEach((row, rowIndex) => {
-//   var rowNumber = rowIndex + 1
-//   var line = document.createElement('div')
-//   line.className = 'row ' + ' row-' + rowNumber
-//   row.forEach((col, colIndex) => {
-//     var colNumber = colIndex + 1
-//     var tile = document.createElement('div')
-//     tile.className = 'tile ' + 'row-' + rowNumber + ' ' + 'col-' + colNumber
-//     tile.id = rowNumber + colNumber
-//     // add special classes for bonus tiles
-//     switch (col) {
-//       case '21':
-//         tile.classList.add('double-letter')
-//         break;
-//       case '31':
-//         tile.classList.add('triple-letter')
-//         break;
-//       case '22':
-//         tile.classList.add('double-word')
-//         break;
-//       case '32':
-//         tile.classList.add('triple-word')
-//         break;
-//       case '99':
-//         tile.classList.add('start-tile')
-//         break;
-//       default:
-//
-//     }
-//     line.appendChild(tile).classList.add(rowNumber + colNumber)
-//   })
-//   board.appendChild(line)
-// })
+// create event listeners on buttons
+document.querySelector('.reset').addEventListener('click', reset)
+document.querySelector('.submit').addEventListener('click', submit)
+document.querySelector('.pass').addEventListener('click', pass)
+
+changePlayer()
+
+function setRack (player) {
+  var rack = document.querySelector('.' + player + '-rack')
+  var currentTiles = Array.from(rack.querySelectorAll('.letter-piece'))
+  if (currentTiles.length < 7) {
+    for (var i = currentTiles.length; i < 7; i++) {
+      var piece = document.createElement('div')
+      piece.textContent = dealTile().letter.toUpperCase()
+      piece.className = 'letter-piece'
+      rack.appendChild(piece)
+    }
+  }
+}
+
+function dealTile () {
+  if (inventory.length > 0) {
+    var piece = sample(inventory)
+    inventory.splice(inventory.indexOf(piece), 1)
```

Comment:
Side effects are not pretty. I'm thinking it would be a good time to introduce testing. Also you may want to use some classes to encapsulate these functions as methods and have an instance reference (`this`) instead of globals.